From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jiaming Zhang <r772577952@gmail.com>
Date: Wed, 9 Sep 2026 14:51:12 +0800
Subject: [PATCH 1/6] sys/syz-extract, prog: update macros and include dirs,
 make elemSize at least 1

update defined macros and include directories in sys/syz-extract, then
we can extract constants in DVB and XFS subsystems.  In mutateSize()
from prog/size.go, make sure the minimal value of elemSize is 1,
otherwise for some inaccurate syscall descriptions (especially for those
missing parameters) the manager may panic.
---
 prog/size.go             | 2 +-
 sys/syz-extract/linux.go | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/prog/size.go b/prog/size.go
index c8f6dfd2e8..0ef1b36d5d 100644
--- a/prog/size.go
+++ b/prog/size.go
@@ -206,7 +206,7 @@ func (r *randGen) mutateSize(arg *ConstArg, parent []Arg, fields []Field) bool {
 						if targetType.Elem.Varlen() {
 							return false
 						}
-						elemSize = targetType.Elem.Size()
+						elemSize = max(targetType.Elem.Size(), 1)
 					}
 				}
 				break
diff --git a/sys/syz-extract/linux.go b/sys/syz-extract/linux.go
index d4de1874a6..3fe15b4dc5 100644
--- a/sys/syz-extract/linux.go
+++ b/sys/syz-extract/linux.go
@@ -142,6 +142,7 @@ func (*linux) processFile(arch *Arch, info *compiler.ConstInfo) (map[string]uint
 		"-O3", // required to get expected values for some __builtin_constant_p
 		"-I.",
 		"-D__KERNEL__",
+		"-D__DVB_CORE__",
 		"-DKBUILD_MODNAME=\"-\"",
 		"-DKBUILD_MODFILE=\"-\"",
 		"-D__LINUX_ARM_ARCH__=7", // arm does not build w/o this
@@ -157,6 +158,8 @@ func (*linux) processFile(arch *Arch, info *compiler.ConstInfo) (map[string]uint
 		"-I" + buildDir + "/include/generated/uapi",
 		"-I" + sourceDir,
 		"-I" + sourceDir + "/include/linux",
+		"-I" + sourceDir + "/fs/xfs/libxfs",
+		"-I" + sourceDir + "/drivers/gpu/drm/i915",
 		"-I" + buildDir + "/syzkaller",
 		"-include", sourceDir + "/include/linux/kconfig.h",
 	}
